src/jrx.c

```diff
@@ -71,7 +71,8 @@ static inline jrx_match_accept _pick_accept(set_match_accept* accepts)
 
         int len = acc.tags[1] - acc.tags[0];
 
-        if ( acc.tags[0] < min || (acc.tags[0] == min && len > min_len) ) {
+        // We take the left-most match. If the new match is at the same offset, prefer the shorter match.
+        if ( acc.tags[0] < min || (acc.tags[0] == min && len < min_len) ) {
```

Comment:
Instead of hardcoding this behavior we should take `REG_FIRST_MATCH`/`JRX_OPTION_FIRST_MATCH` into account. This might require passing the options as an arg here (and also into some callers).